lib/index.js

```diff
@@ -1,3 +1,4 @@
+const path = require('path')
```

Comment:
Introducing `path` as a dependency will make this plugin Node.js only. I'm not planning to use the package in the browser myself, but it's a bit of a shame that others won't be able to just because of this one line.

How about doing something like `/(\\\/)package\.json$/i.test(options.filepath)` to check if we're dealing with `package.json`?

Also adding test for `compose.json`, `package-lock.json` and some `foo-bar.json` will be useful.

Reply:
I’m not fond of the way tests are setup. This is not extensible. Can I introduce jest?

Reply:
Introducing Jest would probably make the scope of this PR too wide. IMHO something like this would be enough for now:


```js
// test/index.js

const prettier = require("prettier")

const input = `...`

const expected = `...`

;["package.json", "Package.json", "package.JSON"].forEach((filepath) => {
  const output = prettier.format(input, {
    filepath,
    plugins: ["."],
  })
  console.assert(output === expected)
})

;["package-lock.json", "compose.json", "foo-bar.json", null].forEach(
  (element) => {
    const output = prettier.format(input, {
      filepath,
      plugins: ["."],
    })
    console.assert(output === input)
  },
)

console.log("\033[32mpassed!\033[39m")
```


P.S.:  [Having no semicolons is painful](https://medium.com/@WebReflection/an-open-letter-to-javascript-leaders-regarding-no-semicolons-82cec422d67d) 😔

Reply:
I added tests as you requested, but with some changes:

- `package.json` is case sensitive. This may not be the case for case insensitive file systems, but `npm` simply doesn’t recognize a file named `Package.json` on my file system.
- Unrecognized JSON files aren’t parsed by prettier using the `json-stringify` parser, also they are formatted differently. I didn’t add tests for those.
- I added some test logging.
- I made the test script actually fail if an assertion failed.